### PR TITLE
Add private ResumeKws, make private SetKeywordActivationState, PauseKws

### PR DESCRIPTION
**Pull request by qfweiwei**
## Purpose

In DialogManager class add private ResumeKws, make private SetKeywordActivationState, PauseKws.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Comment by dargilco** [comment]
I like to good progress!
At this point, please add comments above each one of the public methods in DialogManger.h to describe what the method does. This will be our API documentation.
The following is unclear: What is the difference between  StartListening() and ContinueListening()? Why do we need both? (and an implementation note: internally you should have one call the other, if they share lines of code)

**Comment by dargilco** [comment]
Also think about making sure each one of the public methods behaves well when called out of order or when not relevant (e.g. calling StartKWS when a .table file was not specified). Do we need to define our own set of error codes?

**Comment by dargilco** [comment]
You can handle error codes in a seperate PR if needed.

**Comment by qfweiwei** [comment]
1. Added brief comments above each one of the public methods in DialogManger.h.
2. Make ContinueListening() private, and StartListening() call ContinueListening() to remove duplicate code.
3. Currently, the CPP console allows user to select command at any order. There are no dependencies among commands, and no errors from repeatedly running a command. "4 [start keyword listening]" and "5 [stop keyword listening]" will be invisible if kws is not supported when a .table file was not specified. Therefore, error codes seem not that necessary. A simple and straightforward console app could be better for users to focus on essentials and let them decide what else they need for their scenario. However, if we do need error codes, we need to discuss rules should be enforced and design error codes accordingly.

File: clients/cpp-console/include/DialogManager.h [hunk 1]
```diff
@@ -44,12 +44,17 @@ class DialogManager
     DialogManager(shared_ptr<AgentConfiguration> agentConfig, string audioFilePath);
     const DeviceStatus GetDeviceStatus() { return _deviceStatus; };
     const KeywordActivationState GetKeywordActivationState() { return _keywordActivationState; }
+    //Start a listening session that will terminate after the first utterance. Audio player will be stopped if barge in is supported.
```
Comment (dargilco) [inline clients/cpp-console/include/DialogManager.h:47]: Space after the "//". Same for below

Comment (dargilco) [reply clients/cpp-console/include/DialogManager.h:47]: Regarding "Audio player will be stopped if barge in is supported"  - That's not the right support . TTS playback should stop regardless if barge in is supporet or not. See the [barge in section in the README](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/tree/master/clients#what-does-barge-in-mean) - "Note that in both cases, the user can always press a mic button to trigger a ListenOnceAsync call, which will immediately stop TTS playback and the client will wait for the bot to process the new utterance."

File: clients/cpp-console/include/DialogManager.h [hunk 2]
```diff
@@ -44,12 +44,17 @@ class DialogManager
...
     void StartListening();
+    //Stop audio player, re-initialize back end connection, and restart kws if it is supported.
```
Comment (dargilco) [inline clients/cpp-console/include/DialogManager.h:49]: here you use "kws" and below you use '"keyword recognition". We need to be consistent. I would call it Keyword Spotting to match the KWS in the API name.

File: clients/cpp-console/src/common/DialogManager.cpp [hunk 3]
```diff
@@ -262,14 +261,20 @@ void DialogManager::ResumeKws()
 
 void DialogManager::StartListening()
 {
-    log_t("Now listening...");
     if (_bargeInSupported)
     {
         _player->Stop();
     }
```
Comment (dargilco) [inline clients/cpp-console/src/common/DialogManager.cpp:267]: In all cases stop playback per my comment above. Sorry for not catching that before

File: clients/cpp-console/include/DialogManager.h (hunk 2 above)
Comment (qfweiwei) [reply clients/cpp-console/include/DialogManager.h:49]: Updated.

File: clients/cpp-console/src/common/DialogManager.cpp (hunk 3 above)
Comment (qfweiwei) [reply clients/cpp-console/src/common/DialogManager.cpp:267]: Updated to remove "if (_bargeInSupported)" block.

File: clients/cpp-console/include/DialogManager.h (hunk 1 above)
Comment (qfweiwei) [reply clients/cpp-console/include/DialogManager.h:47]: Updated.

**Pull request merged**